hack/testing/rsyslog/roles/rsyslog/defaults/main.yaml

```diff
@@ -832,8 +832,8 @@ rsyslog__conf_viaq_mmk8s:
           input(type="imfile" file="{{ rsyslog__viaq_log_path }}/*.log" tag="kubernetes" addmetadata="on" reopenOnTruncate="on")
           
           if ((strlen($!CONTAINER_NAME) > 0) and (strlen($!CONTAINER_ID_FULL) > 0)) or
-              ($!metadata!filename startswith "{{ rsyslog__viaq_log_path }}/") then {
-              if $!metadata!filename startswith "{{ rsyslog__viaq_log_path }}/" then {
+              ((strlen($!metadata) > 0) and (strlen($!metadata!filename) > 0) and ($!metadata!filename startswith "/var/log/containers/")) then {
```

Comment:
Does it crash if you omit the `(strlen($!metadata) > 0)`, and only test for `(strlen($!metadata!filename) > 0)`?  Or does it crash if `$!metadata` is unset?
The reason I ask is because I think rsyslog has to actually convert `$!metadata` to string, so the expensive json-to-string functions.  Same with `(strlen($!kubernetes) > 0)` below.

Reply:
> Does it crash if you omit the (strlen($!metadata) > 0), and only test for (strlen($!metadata!filename) > 0)?
Yes, it crashes. :(  I tested with ```(strlen($!metadata!filename) > 0)``` only, first, but it did not solve the crash problem.

Regarding $!kubernetes, I just did copy&paste when I ran into the 2nd crash.  So, I haven't tested without ```(strlen($!kubernetes) > 0)```... :p